### PR TITLE
feat: implementing database with go

### PR DESCRIPTION
**Pull request by tgarg10**

### CODE REVIEW

File: .envrc
Comment (tgarg10) [inline .envrc:1]: This is temporary. We'll be changing thing to use a `config.yaml` instead.

File: README.md [hunk 1]
```diff
@@ -15,8 +15,10 @@ export GOBIN=$GOPATH/bin
 export PATH=$GOPATH/bin:$GOROOT/bin:$PATH
 ```
 5. Install `air` by running `go install github.com/air-verse/air@latest`
-6. On the terminal, run `./dev.sh`
-7. Open http://localhost:8080/ to access the web app.
+6. Install postgres and create a database named `autograder`. Make sure you postgres is up and running. You can download [Beeker Studio](https://www.beekeeperstudio.io/get-community) to you are able to connect to postgres. 
+7. Run `source .envrc` to export the variables of `.envrc` on your local machine. 
+7. On the terminal, run `./dev.sh`
+8. Open http://localhost:8080/ to access the web app.
```
Comment (tgarg10) [inline README.md:21]: This will be overwritten another PR #24

File: db/migrations/20241113210311_create_test.sql
Comment (tgarg10) [inline db/migrations/20241113210311_create_test.sql:1]: This is a test query made for reference.

**Review by fergiedrew — APPROVED** [review]
